tools/create_buffer.py

```diff
@@ -37,12 +56,17 @@ def load_file( file_name ):
             data.append( struct.unpack( 'B', byte )[ 0 ] )
             byte = f.read( 1 )
 
-    return data[ 0 : -1 ]
+    if data[-1] in [0x0a, 0x00]: #['\n', '\0']
+        data[-1] = 0x00
```

Comment:
what about putting \n to the end instead of 0? Wouldn't \n keep wolfssl working without the last-char-ignore-function?

Reply:
It would. By default, there's already a `\n` at the end of this file's buffer. The previous version of the script was chopping it off, so I decided to get rid of it too.

That being said, don't you think it makes sense for this string to be null-terminated?

If the point is to avoid changes to the Wolf BSP, I'm happy to change it. There are a couple other small changes coming soon to that BSP, though.

Reply:
:) right, this is a string for sure. But putting 0 to the end differentiates it from the file version. So memory read will have cert string having 0 at the end, the file read will have \n. And the _prepare_ function will behave differently and we need two of them.
But JC, I leave this up to you since this isn't major issue, I think this is just a question of taste, so up to you, both versions solve the problem well!! Grats!

Reply:
That's a very good point. I'm currently preparing another PR, but I'll update this to terminate with `\n`, and test again later.